app/controllers/course/assessment/submission/answer/answers_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+class Course::Assessment::Submission::Answer::AnswersController < \
+  Course::Assessment::Submission::Answer::Controller
+  load_and_authorize_resource :answer, only: [:create_programming_files, :destroy_programming_file]
+
+  helper Course::Assessment::Submission::SubmissionsHelper.name.sub(/Helper$/, '')
+
+  def create_programming_files
+    if update_answer_files_attributes(@answer, create_programming_files_params)
+      render @answer
+    else
+      render json: { errors: @answer.errors }, status: :bad_request
+    end
+  end
+
+  def destroy_programming_file
+    file_id = delete_programming_file_params[:file_id].to_i
+    if delete_programming_file(@answer, file_id)
+      render json: { answerId: @answer.id, fileId: file_id }
+    else
+      render json: { errors: @answer.errors }, status: :bad_request
+    end
+  end
+
+  private
+
+  def create_programming_files_params
+    params.require(:answer).permit([:id, files_attributes: [:id, :filename, :content]])
+  end
+
+  def delete_programming_file_params
+    params.require(:answer).permit([:id, :file_id])
+  end
+
+  def update_answer_files_attributes(answer, answer_params)
```

Comment:
No need to pass the 2 params, you can use the controller methods and instance variables directly. So something like -> `@answer.specific.create_and_update_files(create_programming_files_params)`